tests/model_serving/model_runtime/vllm/quantization/test_openhermes-2_5_mistral-7b_awq.py

```diff
@@ -0,0 +1,208 @@
+            },
+        ),
+        pytest.param(
+            {"name": "mistral-marlin-raw"},
+            {"model-dir": model_path},
+            {"deployment_type": KServeDeploymentType.RAW_DEPLOYMENT},
+            {
+                "deployment_mode": KServeDeploymentType.RAW_DEPLOYMENT,
+                "runtime_argument": serving_argument,
+                "gpu_count": 1,
+                "quantization": VLLM_SUPPORTED_QUANTIZATION[0],
+                "name": "mistralmar-raw",
+                "min-replicas": 1,
+            },
+        ),
+        pytest.param(
+            {"name": "mistral-awq-serverless"},
+            {"model-dir": model_path},
+            {"deployment_type": KServeDeploymentType.SERVERLESS},
+            {
+                "deployment_mode": KServeDeploymentType.SERVERLESS,
+                "runtime_argument": serving_argument,
+                "gpu_count": 1,
+                "quantization": VLLM_SUPPORTED_QUANTIZATION[1],
+                "name": "mistralawq-ser",
+                "min-replicas": 1,
+            },
+        ),
+        pytest.param(
+            {"name": "mistral-awq-raw"},
+            {"model-dir": model_path},
+            {"deployment_type": KServeDeploymentType.RAW_DEPLOYMENT},
+            {
+                "deployment_mode": KServeDeploymentType.RAW_DEPLOYMENT,
+                "runtime_argument": serving_argument,
+                "gpu_count": 1,
+                "quantization": VLLM_SUPPORTED_QUANTIZATION[1],
+                "name": "mistralawq-raw",
+                "min-replicas": 1,
+            },
+        ),
+    ],
+    indirect=True,
+)
+class TestOpenHermesAWQModel:
+    def test_deploy_model_inference(self, vllm_inference_service, get_pod_name_resource, response_snapshot):
+        if (
+            vllm_inference_service.instance.metadata.annotations["serving.kserve.io/deploymentMode"]
+            == KServeDeploymentType.SERVERLESS
+        ):
+            model_info, chat_responses, completion_responses = fetch_openai_response(
+                url=vllm_inference_service.instance.status.url,
+                model_name=vllm_inference_service.instance.metadata.name,
+            )
+            assert model_info == response_snapshot
+            assert chat_responses == response_snapshot
+            assert completion_responses == response_snapshot
+        if (
+            vllm_inference_service.instance.metadata.annotations["serving.kserve.io/deploymentMode"]
+            == KServeDeploymentType.RAW_DEPLOYMENT
+        ):
+            pod = get_pod_name_resource.name
+            model_details, grpc_chat_response, grpc_chat_stream_responses = run_raw_inference(
+                pod_name=pod, isvc=vllm_inference_service, port=8033, endpoint="tgis"
+            )
+            assert model_details == response_snapshot
+            assert grpc_chat_response == response_snapshot
+            assert grpc_chat_stream_responses == response_snapshot
```

Comment:
duplicate code, please extract to a function and re-use

Reply:
done